cmake/introspection.cmake

```diff
@@ -162,24 +162,6 @@ check_cxx_source_compiles("
   " HAVE_MALLOPT_ARENA_MAX
 )
 
-# Check for posix_fallocate().
```

Comment:
> Ideally this would've been added directly to util so it never had to move.

Yeah...

> Would you mind splitting "cmake: Build bitcoin_util static library" into 2 commits: One that creates util, and another that moves from introspection? That way it's more clear to reviewers and might be more squashable when it comes time to merge upstream.

Would it better to revert partially the "cmake: Check system symbols" commit, and keep the rest of "cmake: Build bitcoin_util static library" as a following commit?

Reply:
Suggested an [update](https://github.com/hebasto/bitcoin/pull/7#issuecomment-1445212086).

Reply:
Hmm, the partial revert looks good, but I'm afraid now this is a bit of a mess.

If we're going to organize this way (which I agree with), I believe a good bit of introspection should be moved down to a more narrow scope. Quick examples:
`HAVE_SYS_PRCTL_H` -> util
`HAVE_DECL_PIPE2` -> util
`HAVE_DECL_SETSID` -> bitcoind
etc.

Reply:
> Hmm, the partial revert looks good, but I'm afraid now this is a bit of a mess.
> 
> If we're going to organize this way (which I agree with), I believe a good bit of introspection should be moved down to a more narrow scope. Quick examples: `HAVE_SYS_PRCTL_H` -> util `HAVE_DECL_PIPE2` -> util `HAVE_DECL_SETSID` -> bitcoind etc.

Did we overdo it a little? Some of symbols are shared between `libbitcoin_util` and `libbitcoinkernel` libraries.